pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -211,14 +216,36 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	}
 
 	if strings.HasPrefix(fi.Name(), ".") {
-		glog.Errorf("Ignoring file: %s", fi.Name())
+		glog.Errorf("Ignoring file (starts with '.'): %s", fi.Name())
 		return nil
 	}
 
+	relative := strings.TrimPrefix(event.Name, w.path)
+	relative = strings.TrimPrefix(relative, "/")
+
 	if !fi.IsDir() {
+		if !strings.HasSuffix(fi.Name(), ".sock") {
```

Comment:
instead of checking name suffix, we can check file mode to make sure it is socket file.